### DISCUSSION TITLE
find value closest to certain mesh location

### DISCUSSION THREAD

**Discussion by Noxical** [usnistgov/fipy#1025] [Q&A] [answered]
Hello, 

I have created a temperature profile over a mesh2D grid. Now I want to get at certain location (point) in the mesh the exact temperature, say X = 4.0 and Y = 2.0:
I know how to do by creating pandas dataframe from the cellValue, see example: 
```
x, y = mesh.cellCenters
df = pd.DataFrame({'x': x, 'y': y, 'T': phi.value})
target_x = 4.8
target_y = 2.2
diff_x = np.abs(df['x'] - target_x)
diff_y = np.abs(df['y'] - target_y)
combined_distance = diff_x + diff_y
closest_row_index = np.argmin(combined_distance)
closest_t_value = df.loc[closest_row_index, 'T']
print(closest_t_value)
```
works like a charm...
However, how would this work in pure fipy? Is it possible or should I just stick to pandas as shown above. 

My prefered option would be using fipy. I was trying something like 
```
T_line = phi.faceValue[mesh.physicalFaces['interface']]
```
However this gives me all the temperatures over certain line (or area if i define cells) but I was not able to extract specific nearest point data from physicalFaces line data. 

Ruud

**Comment by guyer** [comment]
`phi((target_x, target_y))` will give you the value in the cell closest to the point of interest.

`phi((target_x, target_y), order=1)` will give a first-order interpolation of the value at the point of interest.

See [documentation for the `__call__()` method](https://pages.nist.gov/fipy/en/latest/generated/fipy.variables.cellVariable.html#fipy.variables.cellVariable.CellVariable.__call__)

**Comment by Noxical** [comment]
Thanks for your feedback. According the documentation this should work, however I am using a 2Dmesh created with gsmh without uniform grid and I think that is way I get the following error:
```
ValueError: shape mismatch: value array of shape (5042,) could not be broadcast to indexing result of shape (2,)
```
So the cell positions are not ordered according a structured grid and thus interpolation is not straighforward?

**Reply by guyer** [reply, marked as answer]
Sorry, I had the position vector misoriented. `Grid2D` is more tolerant than `Gmsh2D`. Try
```python
phi([[target_x], [target_y]]
```
